File: apps/mobile/modules/breakpoint-ble/android/src/main/java/expo/modules/breakpointble/BreakpointBleModule.kt
package expo.modules.breakpointble

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class BreakpointBleModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("BreakpointBle")

    // Step 2.0 no-op. Real BLE advertise/scan lands in step 2.2.
    Function("hello") {
      "hello from breakpoint-ble (Android)"
    }
  }
}

File: apps/mobile/modules/breakpoint-ble/src/BreakpointBleModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class BreakpointBleModule extends NativeModule<{}> {
  /** Step 2.0 no-op — proves the native module is callable from JS. */
  hello(): string;
}

export default requireNativeModule<BreakpointBleModule>('BreakpointBle');
